cmd/celestia-appd/cmd/migrate_config.go

```diff
@@ -0,0 +1,213 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/v6/app"
+	"github.com/celestiaorg/celestia-app/v6/pkg/appconsts"
+	"github.com/cometbft/cometbft/config"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// ConfigMigrator defines a function type for applying version-specific migrations
+type ConfigMigrator func(*config.Config, *serverconfig.Config) (*config.Config, *serverconfig.Config)
+
+// migrationRegistry maps version strings to their corresponding migration functions
+var migrationRegistry = map[string]ConfigMigrator{
+	"v6": applyV6Migrations,
+}
+
+// migrateConfigCmd returns the migrate-config command that migrates
+// configuration files based on the target version.
+func migrateConfigCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "migrate-config",
+		Short:   "Migrate configuration files to a target version",
+		Long:    "Migrate configuration files (config.toml and app.toml) to be compatible with a target application version.",
+		Example: fmt.Sprintf("celestia-appd migrate-config --home ~/.celestia-app\ncelestia-appd migrate-config --version v%d --home ~/.celestia-app", appconsts.Version),
```

Comment:
[nit] This example can break if we bump `appconsts.Version` to 7 and don't include v7 in the `migrationRegistry` so IMO it would've been sufficient to use v6 here.

```suggestion
		Example: "celestia-appd migrate-config --home ~/.celestia-app\ncelestia-appd migrate-config --version v6 --home ~/.celestia-app",
```